server/auth/auth.go

```diff
@@ -918,12 +933,33 @@ func (auth *AuthManager) handleConnect(conn comms.Link, msg *msgjson.Message) *m
 
 	// Send pending requests for this user.
 	for _, pr := range pendingReqs {
+		// Certain match-related messages should only be sent if the match is
+		// still active.
+		mid, err := pr.req.ExtractMatchID()
+		if err != nil {
+			log.Errorf("Failed to read matchid field from '%s' payload: %v", pr.req.Route, err)
+			// just send it
+		} else if mid != nil && pr.req.Route != msgjson.RevokeMatchRoute {
```

Comment:
Non-nil error is an unmarshal error, but matchid is expected to be nil for payloads with no `"matchid"` (not an error).